### PR TITLE
Auth controller tests

### PR DESCRIPTION
**Pull request by refugedesigns**
### What this PR does (required):
- Mocha tests for all controller routes

### Any information needed to test this feature (required):
- Connect to a test database (Do not use a production database, as the test writes data to the database)
- cd server - and run npm test

### CODE REVIEW

**Review by peterlogo — COMMENTED** [review]
Good job highlighting all the test cases for authenticating a user.

**Comment by moffatethan** [comment]
@refugedesigns I noticed you aren't filling in the PR template again, please as said in your previous PRs. Please fill these in, it's important in a collaborative environment for these to be filled in.

File: client/src/helpers/APICalls/login.ts [hunk 1]
```diff
@@ -1,6 +1,22 @@
-import { AuthApiData } from '../../interface/AuthApiData';
+import { AuthApiData, DemoAuthData } from '../../interface/AuthApiData';
 import { FetchOptions } from '../../interface/FetchOptions';
 
+export const demoLogin = async (email: string, password: string): Promise<DemoAuthData> => {
+  const fetchOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ email, password }),
+  };
```
Comment (moffatethan) [inline client/src/helpers/APICalls/login.ts:9]: We should be setting the base of our branch correctly so we don't see duplicated code.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Tests look good, let's update the base of the branch please so duplicated code is not showing.

File: client/src/helpers/APICalls/login.ts (hunk 1 above)
Comment (refugedesigns) [reply client/src/helpers/APICalls/login.ts:9]: Please what do you mean by updating the base of the branch?

**Comment by refugedesigns** [comment]
> @refugedesigns I noticed you aren't filling in the PR template again, please as said in your previous PRs. Please fill these in, it's important in a collaborative environment for these to be filled in.

for the above pr, the template was filled, I didn't add the screenshot because it is a backend work

File: client/src/helpers/APICalls/login.ts (hunk 1 above)
Comment (moffatethan) [reply client/src/helpers/APICalls/login.ts:9]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please update base of the branch https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Review by moffatethan — APPROVED** [review]

**Pull request merged**